src/crimson/os/seastore/segment_manager/block.cc

```diff
@@ -46,6 +46,38 @@ static write_ertr::future<> do_write(
   });
 }
 
+static write_ertr::future<> do_writev(
+  seastar::file &device,
+  uint64_t offset,
+  bufferlist &bl)
+{
+  logger().debug(
+    "block: do_writev offset {} len {}",
+    offset,
+    bl.length());
+  std::vector<iovec> iov;
+  iov.reserve(bl.get_num_buffers());
+  for (auto& bp : bl.mut_buffers()) {
+    logger().debug("block: do_writev iov length: {}", bp.length());
+    iov.emplace_back(iovec{bp.c_str(), bp.length()});
+  }
+  return device.dma_write(
+    offset,
+    std::move(iov)
+  ).handle_exception([](auto e) -> write_ertr::future<size_t> {
+      logger().error(
+	"do_writev: dma_write got error {}",
+	e);
+      return crimson::ct_error::input_output_error::make();
+  }).then([bl/* hold the buf until the end of io */](auto result)
```

Comment:
```suggestion
  }).then([bl/* hold the buf until the end of io */](size_t written)
```

i'd suggest be explicit unless it hurts readability or flexibility.

Reply:
Done:-)